src/components/ContactDetails/ContactDetailsAvatar.vue

```diff
@@ -0,0 +1,91 @@
+<!--
+import rfcProps from '../../models/rfcProps';
+ * @copyright Copyright (c) 2018 Team Popcorn <teampopcornberlin@gmail.com>
+ *
+ * @author Team Popcorn <teampopcornberlin@gmail.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div :class="{'maximised':maximizeAvatar }" class="contact-header-avatar">
+		<div class="contact-header-avatar__background" @click="toggleSize" />
+		<img v-if="contact.photo" :src="contact.photo"
+			class="contact-header-avatar__picture"
+			@click="toggleSize">
+		<div class="contact-header-avatar__options">
+			<input id="contact-avatar-upload" type="file" class="hidden"
+				accept="image/*" @change="processFile">
+			<label v-tooltip.auto="t('contacts', 'Upload a new picture')" for="contact-avatar-upload"
+				class="icon-upload-white" @click="processFile" />
+			<div v-if="maximizeAvatar" class="icon-delete-white" @click="removePhoto" />
+			<!-- <div v-if="maximizeAvatar" class="icon-fullscreen-white" @click="toggleSize" /> -->
+			<div v-if="maximizeAvatar" class="icon-download-white" @click="downloadPhoto" />
+		</div>
+	</div>
+</template>
+
+<script>
+
+export default {
+	name: 'ContactAvatar',
+
+	props: {
+		contact: {
+			type: Object,
+			required: true
+		}
+	},
+	data() {
+		return {
+			maximizeAvatar: false
+		}
+	},
+	methods: {
+		processFile(event) {
+			let file = event.target.files[0]
```

Comment:
`Cannot read property '0' of undefined` after clicking the input.
So I'm guessing a simple click triggers a change, This is weird :thinking: